include/vcpkg/dependencies.h

```diff
@@ -40,11 +39,30 @@ namespace vcpkg
         EXCLUDED
     };
 
-    struct InstallPlanAction
+    struct BasicAction
     {
-        static bool compare_by_name(const InstallPlanAction* left, const InstallPlanAction* right);
+        static bool compare_by_name(const BasicAction* left, const BasicAction* right);
 
-        InstallPlanAction() noexcept;
+        PackageSpec spec;
+    };
+
+    struct PackageAction : BasicAction
+    {
+        PackageAction(PackageSpec spec, std::vector<PackageSpec> package_dependencies, InternalFeatureSet feature_list)
+            : BasicAction{std::move(spec)}
```

Comment:
This looks like the move/move antipattern and I think `const PackageSpec&` is probably more appropriate. Notably, the caller here is probably not moving into the action so value + move is copy + move which is strictly worse than just copy. Moves of std::strings aren't really that cheap.

I'm not sure on the vector or the InternalFeatureSet but they are likely in a similar situation.

Reply:
Also this should go into the .cpp